### PR TITLE
Replacing matchEntire with find in Regex

### PR DESCRIPTION
**Pull request by neugartf**
****Reason****
Our CI build server happen to return `Picked up JAVA_OPTIONS: -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
javac 1.8.0_222` for `javac -version`, which `matchEntire` doesn't match.

Let me know if I should add tests :)

### CODE REVIEW

**Comment by tschuchortdev** [comment]
Yes, please add a test. I think that function is in need of some testing anyway, but you might have to refactor it a bit since the regex is not independently testable at the moment.

**Comment by neugartf** [comment]
Added two tests. Dont like that refactoring too much, but it works for now. I understand that our problem was an edge case ;)

File: src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt [hunk 1]
```diff
@@ -75,11 +75,16 @@ internal fun getJavacVersionString(javacCommand: String): String {
...
+    return parseVersionString(output, javacCommand)
 }
 
+internal fun parseVersionString(output: String, javacCommand: String) =
+    Regex("javac (.*)?[\\s\\S]*").find(output)?.destructured?.component1()
+        ?: throw IllegalStateException(
+            "Command '$javacCommand -version' did not print expected output. " +
+                    "Output was: '$output'"
+        )
+
```
Comment (tschuchortdev) [inline src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt:87]: I don't like how `parseVersionString` has to know about `javacCommand` just for "logging" purposes. I think it would be better to let the `getJavacVersionString` function handle that:

```kotlin
internal fun getJavacVersionString(javacCommand: String): String {
    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
        .redirectErrorStream(true)
        .start()
    val buffer = Buffer()
    javacProc.inputStream.copyTo(buffer.outputStream())
    javacProc.waitFor()

    val output = buffer.readUtf8()
    return parseVersionString(output) ?: throw IllegalStateException(
            "Command '$javacCommand -version' did not print expected output. " +
                    "Output was: '$output'"
        )
}

internal fun parseVersionString(output: String): String? =
    Regex("javac (.*)?[\\s\\S]*").find(output)?.destructured?.component1()
```

or

```kotlin
internal fun getJavacVersionString(javacCommand: String): String {
    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
        .redirectErrorStream(true)
        .start()
    val buffer = Buffer()
    javacProc.inputStream.copyTo(buffer.outputStream())
    javacProc.waitFor()

    val output = buffer.readUtf8()
    return try {
        parseVersionString(output)
    }
    catch(e: Exception) {
        throw IllegalStateException("Unable to get javac version string from command '$javacCommand -version' ", e)
    }
}

internal fun parseVersionString(javacOutput: String): String =
    Regex("javac (.*)?[\\s\\S]*").find(javacOutput)?.destructured?.component1()
        ?: throw IllegalArgumentException("'$javacOutput' does not contain the expected regex pattern for javac version strings")
```

Comment (neugartf) [reply src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt]: Change the code according to the first suggestion. Try-catch seem to have a bad image in Kotlin :)
https://medium.com/@elizarov/kotlin-and-exceptions-8062f589d07

**Comment by tschuchortdev** [comment]
Thanks for your contribution! I will try to release it this week.
